tap_appstore/streams.py

```diff
@@ -1,72 +1,264 @@
+        th.Property("marketing_opt_in_duration", th.StringType),
+        th.Property("preserved_pricing", th.StringType),
+        th.Property("proceeds_reason", th.StringType),
+        th.Property("consecutive_paid_periods", th.IntegerType),
+        th.Property("paid_service_days_recovered", th.StringType),
+        th.Property("original_start_date", th.DateTimeType),
+        th.Property("client", th.StringType),
+        th.Property("device", th.StringType),
+        th.Property("state", th.StringType),
+        th.Property("country", th.StringType),
+        th.Property("previous_subscription_name", th.StringType),
+        th.Property("previous_subscription_apple_id", th.StringType),
+        th.Property("days_before_canceling", th.StringType),
+        th.Property("cancellation_reason", th.StringType),
+        th.Property("days_canceled", th.IntegerType),
+        th.Property("quantity", th.IntegerType)
     ).to_dict()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def download_data(self, start_date, api):
+        filters = {
+            'frequency': 'DAILY',
+            'reportType': 'SUBSCRIPTION',
+            'reportSubType': 'SUMMARY',
+            'reportDate': start_date,
+            'version': '1_3',
+            'vendorNumber': self.config['vendor_number']
+        }
+        return api.download_sales_and_trends_reports(filters=filters)
+
+
+class FinancialReportStream(client.AppStoreStream):
+    name = "financial_reports"
+    replication_key = "_api_report_date"
+    DATE_FORMAT = '%Y-%m'
+    DATE_INCREMENT = relativedelta(months=1)
+    schema = th.PropertiesList(
+        th.Property("_line_id", th.IntegerType),
+        th.Property("_time_extracted", th.StringType),
+        th.Property("_api_report_date", th.DateTimeType),
+        th.Property("start_date", th.DateTimeType),
+        th.Property("end_date", th.DateTimeType),
+        th.Property("vendor_identifier", th.StringType),
+        th.Property("quantity", th.IntegerType),
+        th.Property("partner_share", th.NumberType),
+        th.Property("extended_partner_share", th.NumberType),
+        th.Property("partner_share_currency", th.StringType),
+        th.Property("sales_or_return", th.StringType),
+        th.Property("apple_identifier", th.StringType),
+        th.Property("title", th.StringType),
+        th.Property("product_type_identifier", th.StringType),
+        th.Property("country_of_sale", th.StringType),
+        th.Property("pre_order_flag", th.StringType),
+        th.Property("promo_code", th.StringType),
+        th.Property("customer_price", th.NumberType),
+        th.Property("customer_currency", th.StringType)
+    ).to_dict()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.date_fields['_api_report_date'] = '%Y-%m'
+        self.skip_line_first_values = ["Total_Rows", "Total_Amount", "Total_Units"]
```

Comment:
Let's move these configs to class variables overwrite from the base class `date_fields_format_map` and `skip_line_first_values`.

Reply:
Moved `skip_line_first_values` but we are no longer using `date_fields_format_map` and are instead setting date fields directly.